### PR TITLE
DNS Swagger description updates

### PR DESCRIPTION
**Pull request by jtuliani**
Update to DNS Swagger descriptions.

Also removes 'If-None-Match' from Zones_Delete and RecordSets_Delete, since If-None-Match is not applicable to delete operations.

---
This change is [Reviewable](https://reviewable.io/reviews/azure/azure-rest-api-specs/658)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jtuliani__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (jonatul). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: arm-dns/2016-04-01/swagger/dns.json [hunk 1]
```diff
@@ -16,13 +16,13 @@
     "application/json"
   ],
   "paths": {
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnszones/{zoneName}/{recordType}/{relativeRecordSetName}": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnsZones/{zoneName}/{recordType}/{relativeRecordSetName}": {
```
Comment (tbombach) [inline arm-dns/2016-04-01/swagger/dns.json]: Why is the capitalization only changed for this one path? This spec has other paths starting with `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnszones/` (no capital 'z'). I think we want to be consistent here

Comment (jtuliani) [inline arm-dns/2016-04-01/swagger/dns.json]: Good catch, I'll fix that...

**Comment by jtuliani** [comment]
@tbombach OK, dnsZones capitalization should be fixed now...pls can you merge?

**Comment by tbombach** [comment]
Merged

**Pull request merged**